### ISSUE TITLE
Unable to find the "window" object for the given node, while testing a mocked modal component

### ISSUE THREAD

**Issue by wesleylhandy** [testing-library/dom-testing-library#658]
- `@testing-library/dom` version: 
    - `@testing-library/jest-dom: ^5.10.1`,
    - `@testing-library/react: ^10.3.0`,
- Testing Framework and version:
    - `jest: ^26.0.1`,
- DOM Environment:

This is with a Gatsby `gatsby: ^2.18.25` project. 

Relevant code or config

### Component Being Tested

```js
const AboutPage = () => {
  const [isModalOpen, setModalOpen] = useState(false)
  return (
    {/*...*/}
    {isModalOpen && (
        <Modal closeClick={() => setModalOpen(false)}></Modal>
    )}
  )
}
```

### Mock of Modal Component

```js
import React from "react"
export default ({ children, ...props }) => (
  <div id="mockModal" data-testid="Modal" {...props}>
    {children}
  </div>
)
```

### Test

```js
import * as React from "react"
import { render, screen, fireEvent } from "@testing-library/react"
import AboutOldPage from "../about-old"
jest.mock(`../../components/modal`)
describe(`AboutOldPage`, () => {
  it(`opens and closes Modal on Button Click`, () => {
    const { getByTestId } = render(<AboutOldPage />)
    
    // initially modal is closed
    let modal = screen.queryAllByTestId(`Modal`)
    expect(modal).toHaveLength(0)

    // open on first button click
    const button = getByTestId(`ToggleModalButton`)
    fireEvent.click(button)
    modal = screen.queryAllByTestId(`Modal`)
    expect(modal).toHaveLength(1)
   
    // close when clicked again
   // THIS IS FAILURE POINT
    fireEvent.click(modal)
    modal = screen.queryAllByTestId(`Modal`)
    expect(modal).toHaveLength(0)
  })
```

What you did:

I have created a mock Modal component and imported it. The modal appears when state changes on button click, and will disappear on button click, but I can't test the modal click. I want to ignore the test since i can test the modal component itself elsewhere, but since I was getting both the error and the message to open an issue and figure there may be something here.

What happened:

<img width="1682" alt="Screen Shot 2020-06-20 at 6 16 49 PM" src="https://user-images.githubusercontent.com/21029295/85212586-3e145000-b322-11ea-8aff-4727182e69a6.png">

Reproduction repository:

https://github.com/wesleylhandy/dom-testing-library-template

```sh
yarn install
yarn test
```

Problem description:

Suggested solution:

**Comment by wesleylhandy** [comment]
I solved my problem. After experimenting again and again with different ways of mocking the components I got a different error that let me to this SO post (https://stackoverflow.com/questions/54468535/how-to-solve-warning-react-does-not-recognize-the-x-prop-on-a-dom-element).

React won't let you spread props on a DOM element. So, instead of spreading props, I created a click handler for the prop I was expecting to be passed down.

So I changed my mock component to

```js
import React from "react"
export default ({ children, ...props }) => (
  <div id="mockModal" data-testid="Modal" onClick={() => props.closeClick()}>
    {children}
  </div>
)
```

This allowed me to test the click event on the Modal that it updated the state on its parent component.
